File: csharp/src/Models/Jewellery.cs
using NecklaceRefactoringKata.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NecklaceRefactoringKata.Models
{
    public record Jewellery(Jewel Stone)
    {
        public virtual bool IsRing() { return false; }
        public virtual bool IsSmall() { return false; }
        public virtual bool IsEarring() { return false; }
        public virtual bool IsNecklace() { return false; }
        public virtual bool IsLarge() { return false; }
    }
}

File: kotlin/src/main/kotlin/org/sammancoaching/necklace/Jewellery.kt
package org.sammancoaching.necklace

open class Jewellery(open val stone: Jewel) {
  open fun isRing(): Boolean = false

  open fun isSmall(): Boolean = false

  open fun isEarring(): Boolean = false

  open fun isNecklace(): Boolean = false

  open fun isLarge(): Boolean = false
}
